src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -276,10 +276,7 @@ ObjectMonitor::ObjectMonitor(oop object) :
 { }
 
 ObjectMonitor::~ObjectMonitor() {
-  if (!_object.is_null()) {
-    // Release object's oop storage if it hasn't already been done.
-    release_object();
-  }
+  release_object();
```

Comment:
I think it would be nice to move the implementation of release_object over to the .cpp file. Alternatively, inline the implementation straight into the destructor. If you don't want to do it in this PR I can do it as a separate change.

Reply:
Done.